pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,280 @@
+package prometheus
```

Comment:
Why name a file `component_conversion_test.go`? why not just `component_test.go` or `component_internal_test.go` (to not trigger external tests linter) ?

Reply:
> Why name a file `component_conversion_test.go`?

Because it mainly has code related to conversion testing.

> why not just `component_test.go` or `component_internal_test.go` 

`component_test.go` already exists.

> (to not trigger external tests linter) ?

I don't understand this :thinking:

Reply:
>I don't understand this thinking

We have a linter, which encourages to use `package foo_test` for test files, so only exported API of package can be tested, as this is considered best practice. You can bypass it by naming file `foo_internal_test.go` instead of `foo_test.go`.

>Because it mainly has code related to conversion testing.

> component_test.go already exists.

I wouldn't split the tests into multiple files. I think if code lives in `component.go`, all related tests should go into `component_test.go` or `component_internal_test.go`, where we should have internal tests as little as possible. I suggest to not invent new convention here.

Reply:
If we decide on moving the test code to be reused by other components. We can move the remaining code to `component_test.go`.

Reply:
>If we decide on moving the test code to be reused by other components.

This is not related IMO. Adding `component_conversion_test.go` opens a way to have all kinds of `component_installation_test.go`, `component_labels_test.go` etc kind of file names and I think we should only have standarized names, so if your code sits in `component.go`, it should ideally only have `component_test.go` and optionally `component_internal_test.go`. Something to put into style guide I guess then...

Reply:
> This is not related IMO. Adding `component_conversion_test.go` opens a way to have all kinds of `component_installation_test.go`, `component_labels_test.go` etc kind of file names and I think we should only have standarized names, so if your code sits in `component.go`, it should ideally only have `component_test.go` and optionally `component_internal_test.go`. Something to put into style guide I guess then...

I would always go for the smaller files than one humongous pile of code in single file. It makes incomprehensible. What is the problem with multiple files? How is it a bad practise to put code in multiple files?

Reply:
If you keep testing code simple, then it doesn't matter how many test cases you have in a single file, as you should always care only about one small function, which misbehaves + a helper in case of TDT. Having a single file make it predictable to where to look for a given test and avoid cluttering the file names spread trough the repo, as I said before.

Reply:
IMO the amount of code in `component_test.go` is not close to being too big so I'd add new tests there.